File: in_go/src/pattern_10.go
package src

import "fmt"

func Pattern_10() {
	row := 9
	column := 9
	symbol := "*"
	halfRow := row / 2

	for i := 0; i < row; i++ {
		for j := 0; j < column; j++ {
			if i <= halfRow {
				if j >= halfRow-i && j <= halfRow+i {
					fmt.Print(string(symbol))
				} else {
					fmt.Print(" ")
				}
			} else {
				tempi := row - 1 - i
				if j >= halfRow-tempi && j <= halfRow+tempi {
					fmt.Print(string(symbol))
				} else {
					fmt.Print(" ")
				}
			}
		}
		fmt.Println()
	}
}

File: in_rust/src/pattern_10.rs
pub fn pattern_10() {
    let row = 9;
    let column = 9;
    let symbol = "*";
    let half_row = row / 2;

    for i in 0..row {
        for j in 0..column {
            if i <= half_row {
                if j >= half_row - i && j <= half_row + i {
                    print!("{}", symbol);
                } else {
                    print!(" ");
                }
            } else {
                let temp_i = row - 1 - i;
                if j >= half_row - temp_i && j <= half_row + temp_i {
                    print!("{}", symbol);
                } else {
                    print!(" ");
                }
            }
        }
        println!();
    }
}
